llvm/lib/Target/AIE/AIEWawRegRewriter.cpp

```diff
@@ -0,0 +1,466 @@
+                         const MachineInstr &DefInstr,
+                         const MachineBasicBlock &MBB);
+
+static bool
+reuseSubRegs(llvm::SmallVector<const MachineInstr *, 4> &OrigInstr,
+             llvm::SmallVector<const MachineInstr *, 4> &OtherInstr);
+
+static bool containEqualSubRegDefs(const MachineInstr *InstrA,
+                                   const MachineInstr *InstrB);
+
+bool AIEWawRegRewriter::runOnMachineFunction(MachineFunction &MF) {
+  if (DisablePass)
+    return false;
+
+  MachineRegisterInfo &MRI = MF.getRegInfo();
+  auto &TRI =
+      *static_cast<const AIEBaseRegisterInfo *>(MRI.getTargetRegisterInfo());
+  VirtRegMap &VRM = getAnalysis<VirtRegMap>();
+  LiveRegMatrix &LRM = getAnalysis<LiveRegMatrix>();
+  LiveIntervals &LIS = getAnalysis<LiveIntervals>();
+  MachineLoopInfo &MLI = getAnalysis<MachineLoopInfo>();
+  bool Modified = false;
+
+  llvm::SmallVector<const MachineBasicBlock *, 4> LoopBBs = getLoopBBs(MLI);
+
+  if (LoopBBs.empty())
+    return false;
+
+  LLVM_DEBUG(llvm::dbgs() << "*** WAW Loop Register Rewriting: "
+                          << MF.getName());
+  LLVM_DEBUG(llvm::dbgs() << " ***\n");
+
+  for (const MachineBasicBlock *MBB : LoopBBs) {
+    LLVM_DEBUG(dbgs() << "WAW Reg Renaming BasicBlock "; MBB->dump();
+               dbgs() << "\n");
+    llvm::SmallVector<MCPhysReg, 16> UsedPhyRegs;
+    llvm::SmallVector<MCPhysReg, 16> UsedVirtRegs;
+    llvm::SmallVector<MCPhysReg, 16> AlreadyReplaced;
+    llvm::SmallVector<std::array<unsigned int, 2>, 8> ReplacementRegs;
+
+    llvm::SmallVector<MCPhysReg, 16> ForbiddenPhysRegs =
+        getBBForbiddenReplacements(MBB, VRM, MRI, &TRI);
+
+    std::map<unsigned, llvm::SmallVector<const MachineInstr *, 4>>
+        UndefPartReuse;
+
+    for (const MachineInstr &MI : *MBB) {
+
+      if (isCopyElimination(MI, VRM))
+        continue;
+
+      for (const MachineOperand &Op : MI.defs()) {
+        Register Reg = Op.getReg();
+        if (!Op.getReg().isVirtual())
+          continue;
+        if (VRM.hasRequiredPhys(Op.getReg()))
+          continue;
+        if (Op.isTied())
+          continue;
+        if (!TRI.isVecOrAccRegClass(MRI.getRegClass(Reg)))
+          continue;
+        // // Subregisters can reuse Constants in the high part of the
+        // // register, therefore renaming a reused physical register can
+        // // prohibit the elimination of a move of constants, thus degrading
+        // // performance if this physical register is removed
+        if (!DisableCopySubRegHandling &&
+            isUndefReserved(Op, MI, *MBB, UndefPartReuse, VRM))
+          continue;
+
+        if (isWAWCandidate(Reg, MI, UsedPhyRegs, UsedVirtRegs, VRM, MRI)) {
+          unsigned int PreSwapPhyReg = VRM.getPhys(Reg).id();
+          bool Replaced = replaceReg(Reg, ForbiddenPhysRegs, AlreadyReplaced,
+                                     MF, VRM, LRM, MRI, LIS, TRI);
+
+          if (Replaced) {
+            MI.dump();
```

Comment:
if this case, we are printing every time we enter this loops. Maybe remove this dump or transform it in debug message, like `LLVM_DEBUG(dbgs() << MI);`

Reply:
fixed